branchandbound.js

```diff
@@ -0,0 +1,143 @@
+var utils = require('./utils')
+
+var maxTries = 1000000
+
+function calculateEffectiveValues (utxos, feeRate) {
+  return utxos.map(function (utxo) {
+    if (isNaN(utils.uintOrNaN(utxo.value))) {
+      return {
+        utxo: utxo,
+        effectiveValue: 0
+      }
+    }
+
+    var effectiveFee = utils.inputBytes(utxo) * feeRate
+    var effectiveValue = utxo.value - effectiveFee
+    return {
+      utxo: utxo,
+      effectiveValue: effectiveValue
+    }
+  })
+}
+
+module.exports = function branchAndBound (utxos, outputs, feeRate) {
+  if (!isFinite(utils.uintOrNaN(feeRate))) return {}
+
+  var costPerOutput = utils.outputBytes({}) * feeRate
+  var costPerInput = utils.inputBytes({}) * feeRate
+  var costOfChange = costPerInput + costPerOutput
+
+  var outAccum = utils.sumOrNaN(outputs)
+
+  var effectiveUtxos = calculateEffectiveValues(utxos, feeRate).filter(function (x) {
+    return x.effectiveValue > 0
+  }).sort(function (a, b) {
+    return b.effectiveValue - a.effectiveValue
+  })
+
+  var selected = search(effectiveUtxos, outAccum, costOfChange)
+  if (selected != null) {
```

Comment:
`standard-js` is already in there.

`!= null` is fine for it, since it checks both undefined and null. Maybe it would be better to somehow return "empty" value from the "inner" function more explicitly (to avoid "null !== undefined" shenanigans), than just `return;`?